engine/src/main/java/org/terasology/rendering/opengl/fbms/DisplayResolutionDependentFBOs.java

```diff
@@ -18,31 +18,46 @@
 import org.lwjgl.opengl.Display;
 import org.terasology.config.RenderingConfig;
 import org.terasology.engine.SimpleUri;
+import org.terasology.engine.subsystem.DisplayDevice;
+import org.terasology.registry.CoreRegistry;
 import org.terasology.rendering.opengl.AbstractFBOsManager;
 import org.terasology.rendering.opengl.FBO;
 import org.terasology.rendering.opengl.FBOConfig;
 import org.terasology.rendering.opengl.ScreenGrabber;
 import org.terasology.rendering.opengl.SwappableFBO;
 
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+
+import static org.terasology.config.RenderingConfig.FBO_SCALE;
+import static org.terasology.engine.subsystem.lwjgl.LwjglDisplayDevice.DISPLAY_RESOLUTION_CHANGE;
 import static org.terasology.rendering.opengl.ScalingFactors.FULL_SCALE;
 
 /**
  * TODO: Add javadocs
  * TODO: Better naming
  */
-public class DisplayResolutionDependentFBOs extends AbstractFBOsManager {
+public class DisplayResolutionDependentFBOs extends AbstractFBOsManager implements PropertyChangeListener {
     public static final SimpleUri FINAL_BUFFER = new SimpleUri("engine:fbo.finalBuffer");
 
     private SwappableFBO gBufferPair;
 
-    private FBO.Dimensions fullScale;
+    private FBO.Dimensions fullScale = new FBO.Dimensions(-1, -1);
     private RenderingConfig renderingConfig;
     private ScreenGrabber screenGrabber;
 
+    private boolean wasTakingScreenshotLastFrame = false;
+
     public DisplayResolutionDependentFBOs(RenderingConfig renderingConfig, ScreenGrabber screenGrabber) {
         this.renderingConfig = renderingConfig;
         this.screenGrabber = screenGrabber;
 
+        renderingConfig.subscribe(FBO_SCALE, this);
```

Comment:
Fair point.

I missed it, because as it turns out, every event is sent to every subscriber either way. But, I added it for the sake of completeness.

Reply:
Wait, scratch that.

renderingConfig is responsible for FBO_SCALE, but NOT for DISPLAY_RESOLUTION_CHANGE.
DISPLAY_RESOLUTION_CHANGE is handled by displayDevice, which is subscribed to separately, in the next line.

Reply:
Oh. I see. I missed it because the subscription to the display class happens without the DISPLAY_RESOLUTION_CHANGE line. Can we add it? Not strictly necessary I guess, but it clarifies what we are interested about.

Reply:
Responded in the latest review.